src/Plugin/EntityBrowser/Widget/View.php

```diff
@@ -199,11 +164,11 @@ public static function processCheckbox(&$element, FormStateInterface $form_state
    * {@inheritdoc}
    */
   public function submit(array &$element, array &$form, FormStateInterface $form_state) {
-    $selected_rows = array_keys(array_filter($form_state->getValue('entity_browser_select')));
+    $selected_rows = array_values(array_filter($form_state->getUserInput()['entity_browser_select']));
```

Comment:
I added an HTML::escape call to each `$row`, which as far as I can tell is all that the Form API would have done.